tests/test_generate.py

```diff
@@ -1865,6 +1867,44 @@ def test_gaussian_noise():
     )  # Bounds are rather liberal to ensure a stable test
 
 
+def test_rotate_horizontal_components():
```

Comment:
Can you add a random rotation? Maybe execute it twice and check that the output is different. Just to make sure this also works.